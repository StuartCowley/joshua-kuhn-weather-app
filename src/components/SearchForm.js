import React from "react";

import "../styles/SearchForm.css";

function SearchForm({ searchText, setSearchText, onSubmit }) {
  const handleInputChange = (event) => setSearchText(event.target.value);

  return (
    <div className="search-form">
      <input
        className="search-form__text"
        type="text"
        placeholder="Enter a city in the UK"
        onChange={handleInputChange}
        value={searchText}
      />
      <button className="search-form__button" type="submit" onClick={onSubmit}>
        Search
      </button>
    </div>
  );
}
export default SearchForm;
