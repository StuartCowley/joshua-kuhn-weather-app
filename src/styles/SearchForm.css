.search-form {
  margin: 3rem;
}

.search-form__button,
.search-form__text {
  margin: 0.25rem;
  padding: 5px;
  color: black;
  font-size: x-large;
}
