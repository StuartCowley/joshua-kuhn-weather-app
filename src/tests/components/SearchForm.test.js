import React from "react";
import { render, screen } from "@testing-library/react";
import SearchForm from "../../components/SearchForm";

describe("SearchForm", () => {
  it("renders correctly", () => {
    const { asFragment } = render(<SearchForm />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("button exists", () => {
    render(<SearchForm />);
    const searchButton = screen.getByRole("button", { name: /Search/i });

    expect(searchButton).toBeInTheDocument();
  });
});
