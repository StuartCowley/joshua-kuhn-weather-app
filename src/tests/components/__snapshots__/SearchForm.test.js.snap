// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<DocumentFragment>
  <div
    class="search-form"
  >
    <input
      class="search-form__text"
      placeholder="Enter a city in the UK"
      type="text"
      value=""
    />
    <button
      class="search-form__button"
      type="submit"
    >
      Search
    </button>
  </div>
</DocumentFragment>
`;
